source/Energinet.DataHub.MeteringPoints.Tests/Repositories/IntegrationEventRepositoryTests.cs

```diff
@@ -0,0 +1,136 @@
+using Energinet.DataHub.MeteringPoints.Infrastructure.DataAccess;
+using Energinet.DataHub.MeteringPoints.Infrastructure.Helpers;
+using Energinet.DataHub.MeteringPoints.Infrastructure.IntegrationServices.Helpers;
+using Energinet.DataHub.MeteringPoints.Infrastructure.IntegrationServices.Repository;
+using Energinet.DataHub.MeteringPoints.Infrastructure.Outbox;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+using Xunit;
+using Xunit.Categories;
+
+namespace Energinet.DataHub.MeteringPoints.Tests.Repositories
+{
+    [UnitTest]
+    public class IntegrationEventRepositoryTests
+    {
+        private readonly DbContextOptions<MeteringPointContext> _options;
+        private readonly IJsonSerializer _jsonSerializer;
+        private IIntegrationEventRepository _integrationEventRepository;
+
+        public IntegrationEventRepositoryTests()
+        {
+            _options = new DbContextOptionsBuilder<MeteringPointContext>()
+                .UseInMemoryDatabase(databaseName: "OutboxMessages")
+                .Options;
+            _jsonSerializer = new JsonSerializer();
+        }
+
+        [Fact]
+        public async Task IntegrationEventRepositoryGetUnProcessedIntegrationEventMessageAsyncTest()
+        {
+            await using var context = new MeteringPointContext(_options);
+            context.Add(CreateOutBoxMessage(Guid.NewGuid()));
+
+            await context.SaveChangesAsync().ConfigureAwait(false);
+
+            _integrationEventRepository = new IntegrationEventRepository(context, _jsonSerializer);
+            var outBoxMessage = await _integrationEventRepository.GetUnProcessedIntegrationEventMessageAsync().ConfigureAwait(false);
+
+            Assert.IsType<OutboxMessage>(outBoxMessage);
+            Assert.Equal("CreateMeteringPointEventMessage", outBoxMessage.Type);
+        }
+
+        [Fact]
+        public async Task IntegrationEventRepositoryMarkIntegrationEventMessageAsProcessedAsyncTest()
+        {
+            var guid = Guid.NewGuid();
+            await using var context = new MeteringPointContext(_options);
+
+            context.Add(CreateOutBoxMessage(guid));
+            await context.SaveChangesAsync().ConfigureAwait(false);
+
+            _integrationEventRepository = new IntegrationEventRepository(context, _jsonSerializer);
+
+            await _integrationEventRepository.MarkIntegrationEventMessageAsProcessedAsync(guid)
+                .ConfigureAwait(false);
+            await context.SaveChangesAsync().ConfigureAwait(false);
+
+            var entity = await context.OutboxMessages.FindAsync(guid).ConfigureAwait(false);
+            Assert.True(entity.ProcessedDate != null);
+        }
+
+        [Fact]
+        public async Task IntegrationEventRepositorySaveIntegrationEventMessageToOutboxAsyncTest()
+        {
+            await using var context = new MeteringPointContext(_options);
+
+            CreateMeteringPointEventMessage message = new(
+                "12345678",
```

Comment:
We should use a valid gsrn so it can be pass yet to be implemented validations. Eg. (571234567891234605)
```suggestion
                "571234567891234605",
```

Reply:
Fair enough :)